packages/dapp-connector/src/connector.js

```diff
@@ -0,0 +1,140 @@
+  // https://github.com/facebook/hermes/issues/612
+  'babel plugin show source'
+  if (typeof iconUrl !== 'string') throw new Error('iconUrl must be a string')
+  if (typeof apiVersion !== 'string') throw new Error('apiVersion must be a string')
+  if (typeof walletName !== 'string') throw new Error('walletName must be a string')
+  if (!Array.isArray(supportedExtensions)) throw new Error('supportedExtensions must be an array')
+
+  if (window.cardano && window.cardano[walletName]) return
+
+  const postMessage = (data) => {
+    if (window.ReactNativeWebView) {
+      window.ReactNativeWebView.postMessage(JSON.stringify(data))
+      return
+    }
+    window.postMessage(JSON.stringify(data))
+  }
+  const promisesMap = new Map()
+
+  const getRandomId = () => Math.random().toString(36).substr(2, 9)
+
+  const callExternalMethod = (method, args, options = {}) => {
+    const requestId = getRandomId()
+
+    if (options?.doNotWaitForResponse) {
+      postMessage({id: requestId, method, params: {args, browserContext: getContext()}})
+      return Promise.resolve()
+    }
+    const promise = new Promise((resolve, reject) => {
+      promisesMap.set(requestId, {resolve, reject})
+    })
+    const params = {args, browserContext: getContext()}
+    postMessage({id: requestId, method, params})
+    return promise
+  }
+
+  const logMessage = (...args) => {
+    callExternalMethod('log_message', args, {doNotWaitForResponse: true})
+  }
+
+  window.addEventListener('error', (event) => {
+    logMessage('Unhandled error:' + serializeError(event.error))
+  })
+
+  window.addEventListener('unhandledrejection', (event) => {
+    logMessage('Unhandled rejection:' + serializeError(event.reason))
+  })
+
+  const serializeError = (error) => {
+    if (error instanceof Error) {
+      return error.name + ': ' + error.message
+    }
+    return JSON.stringify(error)
+  }
+
+  window.addEventListener('message', (event) => {
+    logMessage('Received message' + JSON.stringify(event.data))
+    const {id, result, error} = event.data
+    const promise = promisesMap.get(id)
+    if (!promise) return
+    promisesMap.delete(id)
+    if (error) {
+      promise.reject(handleError(error))
+    } else {
+      promise.resolve(result)
+    }
+  })
+
+  const createApi = (cardanoEnableResponse) => {
+    if (!cardanoEnableResponse) {
+      logMessage('User Rejected')
+      throw {code: -3, info: 'User Rejected'}
+    }
+    localStorage.setItem('yoroi-session-id', sessionId)
+
+    enabling = false
+    return {
+      getBalance: (...args) => callExternalMethod('api.getBalance', args),
+      getChangeAddress: (...args) => callExternalMethod('api.getChangeAddress', args),
+      getNetworkId: (...args) => callExternalMethod('api.getNetworkId', args),
+      getRewardAddresses: (...args) => callExternalMethod('api.getRewardAddresses', args),
+      getUsedAddresses: (...args) => callExternalMethod('api.getUsedAddresses', args),
+    }
+  }
```

Comment:
@michaeljscript can we add JS Docs
/**
 * @typedef {Object} ExternalAPI
 * @property {(...args: any[]) => Promise<any>} getBalance Function to get balance.
 * @property {(...args: any[]) => Promise<any>} getChangeAddress Function to get change address.
 * @property {(...args: any[]) => Promise<any>} getNetworkId Function to get network ID.
 * @property {(...args: any[]) => Promise<any>} getRewardAddresses Function to get reward addresses.
 * @property {(...args: any[]) => Promise<any>} getUsedAddresses Function to get used addresses.
 */

/**
 * @type {ExternalAPI}
 */

This will make any IDE, to know about the types.